pkg/controller/sidecar/sync.go

```diff
@@ -104,6 +105,39 @@ func (c *Controller) decommissionNode(ctx context.Context, svc *corev1.Service)
 	return nil
 }
 
+func (c *Controller) setHostIdLabel(ctx context.Context, svc *corev1.Service) error {
```

Comment:
nit: setter methods don't normally fetch the value inside but take it as an argument